R/COPD-Derived.R

```diff
@@ -0,0 +1,196 @@
+  function(DHHGAGE_cont, CCC_91E, CCC_91F) {
+    # Argument verification
+    if ((!is_equal(CCC_91E, 1) &
+         !is_equal(CCC_91E, 2)) |
+        (!is_equal(CCC_91F, 1) &
+         !is_equal(CCC_91F, 2))) {
+      warning(
+        paste(
+          "In DHHGAGE_cont:",
+          DHHGAGE_cont,
+          ", CCC_91E:",
+          CCC_91E,
+          ", CCC_91F:",
+          CCC_91F,
+          "one or more of the arguments was outside the 1:2 allowed
+          range however the condition is still calculated",
+          sep = ""
+        ), call. = FALSE
+      )
+    }
+    if_else2(
+      (DHHGAGE_cont > 0 & DHHGAGE_cont >= 35) &
+        (CCC_91E == 1 | CCC_91F == 1), 1,
+    if_else2(
+      ((DHHGAGE_cont > 0 & DHHGAGE_cont < 35) &
+        (CCC_91E == 1) | CCC_91F == 1), 2,
+    if_else2(
+      ((DHHGAGE_cont > 0 & DHHGAGE_cont < 35) &
+        (CCC_91E == 2 & CCC_91F == 2)), 3,
+    if_else2(
+      ((DHHGAGE_cont > 0 & DHHGAGE_cont >= 35) &
+        (CCC_91E == 2 & CCC_91F == 2 )), 3, "NA(b)"
+                )
+          )
+      )
+    )
+  }
+
+#' @title COPD_Emph_der_fun2
+#'
+#' @description This is one of 2 functions used to create a derived variable
+#'  (COPD_Emph_der) that determines if a respondents has either COPD or 
+#'  Emphysema. 2 different functions have been created to account for the fact
+#'  that different respiratory variables are used across CCHS cycles. This
+#'  function is for CCHS cycles (2001-2003, 2009-2014) that use COPD and Emphysema as
+#'  a combined variable. 
+#'
+#' @param DHHGAGE_cont continuous age variable.
+#'
+#' @param CCC_091 variable indicating if respondent has either COPD or Emphysema
+#'
+#' @return a categorical variable (COPD_Emph_der) with 3 levels:
+#'
+#'  \enumerate{
+#'  \item respondent is over the age of 35 and has a respiratory condition
+#'  \item respondent is under the age of 35 and has a respiratory condition
+#'  \item respondent does not have a respiratory condition
+#'  }
+#'
+#' @examples
+#' # COPD_Emph_der_fun2() to create values across CCHS cycles
+#' # (2001-2003, 2009-2014) COPD_Emph_der_fun2() is specified in
+#' # variable_details.csv along with the CCHS variables and cycles included.
+#'
+#' # To transform COPD_Emph_der, use rec_with_table() for each CCHS cycle
+#' # and specify COPD_Emph_der, along with the various respiratory
+#' # variables. Then by using bind_rows() you can combine COPD_Emph_der
+#' # across cycles.
+#'
+#' library(cchsflow)
+#'
+#' COPD2001 <- suppressWarnings(rec_with_table(
+#'   cchs2001_p,  c(
+#'     "DHHGAGE_cont", "CCC_091",
+#'     "COPD_Emph_der"
+#'   )
+#' ))
+#'
+#' head(COPD2001)
+#'
+#' COPD2014 <- suppressWarnings(rec_with_table(
+#'   cchs2007_2008_p, c(
+#'     "DHHGAGE_cont", "CCC_091", 
+#'     "COPD_Emph_der"
+#'   )
+#' ))
+#'
+#' tail(COPD2014)
+#'
+#' combined_COPD <- suppressWarnings(bind_rows(COPD2001, COPD2014))
+#'
+#' head(combined_COPD)
+#' tail(combined_COPD)
+#' @seealso \code{\link{COPD_Emph_der_fun2}}
```

Comment:
Should be `@seealso \code{\link{COPD_Emph_der_fun1}}`

Reply:
How to handle different NAs will vary somewhat base on the question. See discussion for ADL. https://github.com/Big-Life-Lab/cchsflow/pull/74 Oct. 28 comment.